performance/cinterop/src/main/kotlin/main.kt

```diff
@@ -21,29 +21,26 @@ import org.jetbrains.structsBenchmarks.*
 import org.jetbrains.typesBenchmarks.*
 import org.jetbrains.kliopt.*
 
-class CinteropLauncher(numWarmIterations: Int, numberOfAttempts: Int, prefix: String) : Launcher(numWarmIterations, numberOfAttempts, prefix) {
-    val stringBenchmark = StringBenchmark()
-    val intMatrixBenchmark = IntMatrixBenchmark()
-    val intBenchmark = IntBenchmark()
-    val boxedIntBenchmark = BoxedIntBenchmark()
+class CinteropLauncher : Launcher() {
     override val benchmarks = BenchmarksCollection(
             mutableMapOf(
-                    "macros" to ::macrosBenchmark,
-                    "struct" to ::structBenchmark,
-                    "union" to ::unionBenchmark,
-                    "enum" to ::enumBenchmark,
-                    "stringToC" to stringBenchmark::stringToCBenchmark,
-                    "stringToKotlin" to stringBenchmark::stringToKotlinBenchmark,
-                    "intMatrix" to intMatrixBenchmark::intMatrixBenchmark,
-                    "int" to intBenchmark::intBenchmark,
-                    "boxedInt" to boxedIntBenchmark::boxedIntBenchmark
+                    "macros" to FunctionBenchmarkEntry(::macrosBenchmark),
+                    "struct" to FunctionBenchmarkEntry(::structBenchmark),
+                    "union" to FunctionBenchmarkEntry(::unionBenchmark),
+                    "enum" to FunctionBenchmarkEntry(::enumBenchmark),
+                    "stringToC" to InstanceBenchmarkEntry.create(::StringBenchmark, { stringToCBenchmark() }),
```

Comment:
FunctionBenchmarkEntry/InstanceBenchmarkEntry doesn't look too descriptive.
Maybe just  BenchmarkEntry with two different constructors? Or BenchmarkEntryWithInit?

Reply:
I have different lambdas types and I need to separate them, may be then AbstractBenchmarkEntry for interface and BenchmarkEntry and BenchmarkEntryWithInit for children?